### PR TITLE
[tt-6671]  AWS Kinesis pump

### PR DESCRIPTION
**Pull request by joshblakeley**
## Description
This PR adds a Tyk pump for AWS Kinesis pubsub broker.

Example config is simple:

```
 "kinesis":{
      "type": "kinesis",
      "meta": {
        "stream_name": "tyktest",
        "region": "eu-central-1"
      }
    }
```

Other fields are batch_size (Default 100 - limit is 500)

## Related Issue
https://tyktech.atlassian.net/browse/TT-6671

## Motivation and Context

Customer base requirements

## How This Has Been Tested
Tested on AWS with Kinesis at load.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [x] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

File: pumps/kinesis.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+// WriteData writes the analytics records to AWS Kinesis in batches.
+func (p *KinesisPump) WriteData(ctx context.Context, records []interface{}) error {
+	batches := splitIntoBatches(records, p.kinesisConf.BatchSize)
+	for _, batch := range batches {
+		var entries []types.PutRecordsRequestEntry
+		for _, record := range batch {
+			// Build message format
+			decoded := record.(analytics.AnalyticsRecord)
```
Comment (tbuchaillot) [inline pumps/kinesis.go:99]:
```go
decoded, ok := record.(analytics.AnalyticsRecord)
if !ok {
  p.log.WithField("record", record).Error("unable to decode record")
  continue
}
```

File: pumps/kinesis.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			}
+			entries = append(entries, entry)
+		}
+
+		input := &kinesis.PutRecordsInput{
+			StreamName: aws.String(p.kinesisConf.StreamName),
+			Records:    entries,
+		}
+
+		output, err := p.client.PutRecords(context.TODO(), input)
```
Comment (tbuchaillot) [inline pumps/kinesis.go:143]: We should use the context that's received in `WriteData`

```go
		output, err := p.client.PutRecords(ctx, input)
```

File: pumps/kinesis.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+// WriteData writes the analytics records to AWS Kinesis in batches.
+func (p *KinesisPump) WriteData(ctx context.Context, records []interface{}) error {
+	batches := splitIntoBatches(records, p.kinesisConf.BatchSize)
+	for _, batch := range batches {
+		var entries []types.PutRecordsRequestEntry
+		for _, record := range batch {
+			// Build message format
+			decoded := record.(analytics.AnalyticsRecord)
+			message := Json{
```
Comment (tbuchaillot) [inline pumps/kinesis.go:100]: Try changing name of var or `//nolint:dupl`

**Review by mativm02 — APPROVED** [review]
lgtm!

**Pull request merged**